editor/components/post-taxonomies/hierarchical-term-selector.js

```diff
@@ -242,10 +242,14 @@ class HierarchicalTermSelector extends Component {
 		const newTermSubmitLabel = newTermButtonLabel;
 		const inputId = `editor-post-taxonomies__hierarchical-terms-input-${ instanceId }`;
 
+		if ( ! hasAssignAction ) {
```

Comment:
Same as in `flat-term-selector.js`, this check could happen earlier.

Reply:
ac803ad85dc5dfd2cbe70a92f5fb09cf2483deb7